scripts/test

```diff
@@ -20,20 +24,29 @@ if [ ${ARCH} == arm64 ]; then
     export ETCD_UNSUPPORTED_ARCH=arm64
 fi
 
-echo Starting rancher server
+echo Starting rancher server for test
 touch /tmp/rancher.log
-./scripts/run >/tmp/rancher.log 2>&1 &
-PID=$!
-trap cleanup exit
 
-keepalive()
+run_rancher()
 {
     while sleep 2; do
-        if [ ! -e /proc/$PID ]; then
+        if [ "$PID" != "-1" ] && [ ! -e /proc/$PID ]; then
             echo Rancher died
-            cat /tmp/rancher.log
-            ./scripts/run >/tmp/rancher.log 2>&1 &
-            PID=$!
+            echo Rancher logs were
+            tail -n 25 /tmp/rancher.log
+            echo K3s logs were:
+            tail -n 25 build/testdata/k3s.log
+            if [ "$INT_TESTS_STARTED" = "true" ]; then
+              echo Rancher died after tests started, aborting
+              exit 1
+            fi
+            PID=-1
+            sleep 5
+        fi
+        if [ "$PID" = "-1" ]; then
+          echo Starting rancher server using run
+          ./scripts/run >/tmp/rancher.log 2>&1 &
+          PID=$!
```

Comment:
Is it possible to be trapped in the loop of "restarting Rancher -> process dies before starting tests -> restarting rancher -> process dies before starting tests --> restarting Rancher ..." 

If so, we may force it to exit & fail if a max retry count is hit.

Reply:
If that is the case, Drone will eventually kill off the job after a timeout interval is hit

I didn't want to artificially limit the amount of startup attempts. That would maybe be a good idea for a future enhancement though

Reply:
Oh, nice. It is good to know that Drone will handle it.